drivers/mge-hid.c

```diff
@@ -227,7 +227,7 @@ static const char *eaton_abm_enabled_fun(double value)
 {
 	advanced_battery_monitoring = value;
 
-	if (advanced_battery_monitoring == ABM_ENABLED_TYPE)
+	if (dstate_getinfo("battery.charger.type"))
```

Comment:
+1 as with my other reservations about the variable existence approach deciding on which ABM mode is in effect, if `ABM_ENABLED_TYPE` does not work for the comparison here perhaps it being `4` is not the correct value?